### PR TITLE
Improve password generation for link shares

### PR DESCRIPTION
**Pull request by PVince81**
Use web crypto when generating password for link shares whenever the password policy app is disabled.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included => none written also no space to add any for such files
- [x] Screenshots before/after for front-end changes => not relevant
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required => n/a
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes) => unsure

### CODE REVIEW

File: apps/files_sharing/src/utils/GeneratePassword.js [hunk 1]
```diff
@@ -49,10 +50,12 @@ export default async function() {
...
-			return prev
-		}, '')
+	const array = new Uint8Array(10)
+	const ratio = passwordSet.length / 255
+	self.crypto.getRandomValues(array)
+	let password = ''
+	for (let i = 0; i < array.length; i++) {
+		password += passwordSet.charAt(array[i] * ratio)
+	}
+	return password
```
Comment (artonge) [inline apps/files_sharing/src/utils/GeneratePassword.js:60]: Maybe easier to understand:

```suggestion
return Array.from(self.crypto.getRandomValues(new Uint8Array(10)))
	.map(x => x % passwordSet.length)
	.map(x => passwordSet.charAt(x))
	.join('')
```

**Review by artonge — APPROVED** [review]
Nitpick, but good in any case

File: apps/files_sharing/src/utils/GeneratePassword.js (hunk 1 above)
Comment (ChristophWurst) [reply apps/files_sharing/src/utils/GeneratePassword.js:60]: Nitpick²: `.map(passwordSet.charAt)` :exploding_head:

**Review by ChristophWurst — APPROVED** [review]

File: apps/files_sharing/src/utils/GeneratePassword.js (hunk 1 above)
Comment (PVince81) [reply apps/files_sharing/src/utils/GeneratePassword.js:60]: ah! I had tried it before without `Array.from`

but also the proposal is doing two passes on the array while mine only does one

let's leave it for now

Comment (PVince81) [reply apps/files_sharing/src/utils/GeneratePassword.js:60]: actually three passes as Array.from might do another copy :-D

**Pull request merged**